triton-client/src/components/DatasetColumns.tsx

```diff
@@ -0,0 +1,371 @@
+import { Button, Modal, notification, Spin } from "antd"
+import { InfoCircleOutlined } from "@ant-design/icons"
+import { ReactNode, useCallback, useMemo, useState } from "react"
+import { CloseCircleOutlined, PlusCircleOutlined } from "@ant-design/icons"
+import { DownloadRequestType, TritonDataset } from "@api/api-types"
+import { useAppDispatch, useAppSelector } from "@store/hooks"
+import {
+    deleteDownloadRequest,
+    createDownloadRequest,
+    extendStagingRequest,
+} from "@store/thunks"
+import { selectConstants } from "@store/constants"
+import { ActionDropdownProps } from "@components/ActionDropdown/interfaces"
+import { ActionDropdown, ReadsPerSample } from "@components/."
+import { Provider } from "react-redux"
+import { store } from "@store/store"
+import config from "@common/config"
+import { ColumnsType } from "antd/es/table"
+import { dataSize } from "@common/functions"
+import { RequestState } from "@store/requests"
+
+export interface DatasetColumnSource {
+    id: TritonDataset["id"]
+    lane: TritonDataset["lane"]
+    external_project_id: TritonDataset["external_project_id"]
+    latest_release_update: Date
+    isFetchingRequest: boolean
+    activeRequest: RequestState | undefined
+    totalSize: number
+}
+
+export function useDatasetColumns(
+    updateDataset: (datasetID: TritonDataset["id"]) => Promise<void>,
+) {
+    return useMemo(() => {
+        const columns: ColumnsType<DatasetColumnSource> = []
+        columns.push({
+            title: "",
+            dataIndex: "id",
+            key: "metric",
+            width: "1%",
```

Comment:
manually assigning css related values at this level is not good frontend dev practice. its good to have the styling centralized at either the each level of interest, (in the index.scss file, global.scss, etc)